toolchain/check/name_lookup.h

```diff
@@ -99,6 +99,13 @@ auto LookupQualifiedName(Context& context, SemIR::LocId loc_id,
 auto LookupNameInCore(Context& context, SemIR::LocId loc_id,
                       llvm::StringRef name) -> SemIR::InstId;
 
+// Checks whether a name is accessible in the given access context. Produces a
+// diagnostic if not.
+auto CheckAccess(Context& context, SemIR::LocId loc_id,
```

Comment:
```suggestion
auto RequireAccess(Context& context, SemIR::LocId loc_id,
```

We use "Require" in other places to say "make sure this is true or diagnose it" like RequireCompleteType for example

Reply:
"RequireAccessible" would work better for me than "RequireAccess", but both feel a bit off to me. Maybe "RequireAccessibleMember"? Though we do also have a lot of "Check" functions that mean "check that this satisfies the semantic constraints or diagnose it", and it looks like we have more of those than we have of the "Require" family -- 49 hits for `/auto Check[A-Z][A-Za-z]*\(/` versus 15 for `/auto Require[A-Z][A-Za-z]*\(/ -- so I'm inclined to use "Check" here too for consistency with that.